src/features/environmentDetails/components/Specification/SpecificationEdit.tsx

```diff
@@ -116,7 +142,11 @@ export const SpecificationEdit = ({
       dispatch(updatePackages(code.dependencies));
       dispatch(updateChannels(code.channels));
     } else {
-      setCode({ dependencies: requestedPackages, channels });
+      setCode({
```

Comment:
So I tested this manually and now I believe pretty strongly that we should fix the asymmetry I mentioned before and dispatch the `updateEnvironmentVariables` action when the user toggles from YAML to GUI mode, just as we do for channels and dependencies; otherwise, the user loses the changes they made in YAML mode.

Steps to recreate. Go to an environment, click edit. Toggle into YAML mode. Change the `variables` field. Toggle into GUI mode. Toggle back into YAML mode. Notice that the changes you made to the `variables` field are undone.

Reply:
> Steps to recreate. Go to an environment, click edit. Toggle into YAML mode. Change the variables field. Toggle into GUI mode. Toggle back into YAML mode. Notice that the changes you made to the variables field are undone.

Oh, that's a good point! I had no idea it works like that. I thought the whole reason we have the Save button there is to save the changes. But yeah, I've just tried this with dependencies and it does update them in the GUI simply if you switch back and forth between YAML and GUI without clicking on Save.